typescript/sdk/src/deploy/proxy.ts

```diff
@@ -135,3 +135,44 @@ export function proxyAdminUpdateTxs(
 
   return transactions;
 }
+
+/**
+ * Check if contract bytecode matches ProxyAdmin patterns
+ * This is more efficient than function calls but less reliable
+ * @param provider The provider to use
+ * @param address The contract address
+ * @returns true if the bytecode suggests it's a ProxyAdmin
+ */
+export async function isProxyAdminByBytecode(
+  provider: EthersLikeProvider,
+  address: Address,
+): Promise<boolean> {
+  try {
+    const code = await provider.getCode(address);
+    if (code === '0x') {
+      return false;
+    }
+
+    // Check for common ProxyAdmin function selectors in bytecode
+    const proxyAdminSelectors = [
+      '0x8da5cb5b', // owner()
+      '0xf3b7dead', // getProxyAdmin(address)
+      '0x204e1c7a', // getProxyImplementation(address)
+      '0x99a88ec4', // upgrade(address,address)
+      '0x9623609d', // upgradeAndCall(address,address,bytes)
+      '0x7eff275e', // changeProxyAdmin(address,address)
```

Comment:
is there a way with ethers to ge tthe function selector by using its name? so we avoid hardcoding the signature, if not no problem